src/Conference.ts

```diff
@@ -162,30 +169,39 @@ export class Conference {
-                        this.auditoriums[createEvent[RSC_AUDITORIUM_ID]] = new Auditorium(room, this.client, this);
-                        break;
-                    case RoomKind.AuditoriumBackstage:
-                        this.auditoriumBackstages[createEvent[RSC_AUDITORIUM_ID]] = new AuditoriumBackstage(room, this.client, this);
-                        break;
-                    case RoomKind.Talk:
-                        this.talks[createEvent[RSC_TALK_ID]] = new Talk(room, this.client, this);
-                        break;
-                    case RoomKind.SpecialInterest:
-                        this.interestRooms[createEvent[RSC_SPECIAL_INTEREST_ID]] = new InterestRoom(room, this.client, this);
-                        break;
-                    default:
-                        break;
-                }
-            }
+        const roomIds = await this.client.getJoinedRooms();
+        const batchSize = 20;
+        for (let i = 0; i < roomIds.length; i += batchSize) {
+            // Process batches of rooms in parallel, since there may be a few hundred
+            const tasks = roomIds.slice(i, i + batchSize).map(
+                roomId => (
+                    async function(client: MatrixClient, conference: Conference, roomId: string) {
+                        const createEvent = await client.getRoomStateEvent(roomId, "m.room.create", "");
+                        if (createEvent[RSC_CONFERENCE_ID] === conference.id) {
+                            switch (createEvent[RSC_ROOM_KIND_FLAG]) {
+                                case RoomKind.Conference:
+                                    conference.dbRoom = new MatrixRoom(roomId, client, conference);
+                                    break;
+                                case RoomKind.Auditorium:
+                                    conference.auditoriums[createEvent[RSC_AUDITORIUM_ID]] = new Auditorium(roomId, client, conference);
+                                    break;
+                                case RoomKind.AuditoriumBackstage:
+                                    conference.auditoriumBackstages[createEvent[RSC_AUDITORIUM_ID]] = new AuditoriumBackstage(roomId, client, conference);
+                                    break;
+                                case RoomKind.Talk:
+                                    conference.talks[createEvent[RSC_TALK_ID]] = new Talk(roomId, client, conference);
+                                    break;
+                                case RoomKind.SpecialInterest:
+                                    conference.interestRooms[createEvent[RSC_SPECIAL_INTEREST_ID]] = new InterestRoom(roomId, client, conference);
+                                    break;
+                                default:
+                                    break;
+                            }
+                        }
+                    }
+                )(this.client, this, roomId)
+            )
```

Comment:
the linter in this project is broken/non-existent, but would be good to follow the Element Web practices where feasible. In many cases this isn't already the case (it uses Travis Style™), but should be pretty close.

Most importantly for these lines though:
```suggestion
                )(this.client, this, roomId);
            );
```

Reply:
Good spot! I think only the 2nd semicolon is missing. The first would be inside the arguments of a big `.map(...)` call.

I'm not familiar with Element Web style so you might have to spell anything else out to me.


I really want to bring in a linter and turn on typescript `strictNullChecks`, but ensuring that everything works for FOSDEM is the priority for now.